spanner/transaction.go

```diff
@@ -853,6 +890,33 @@ func (t *ReadOnlyTransaction) release(err error) {
 	}
 }
 
+func errCannotSetDirectedReadOptions() error {
+	return spannerErrorf(codes.InvalidArgument, "DirectedReadOptions cannot be set for ReadWriteTransaction or PDML")
+}
+
+func errInvalidLenDirectedReadOptions() error {
+	return spannerErrorf(codes.InvalidArgument, "Maximum length of replica selection allowed in DirectedReadOptions is 10")
+}
+
+func errInvalidDirectedReadOptions() error {
+	return spannerErrorf(codes.InvalidArgument, "Only one of IncludeReplicas or ExcludeReplicas can be set")
+}
+
+func (t *ReadOnlyTransaction) validateDirectedReadOptions(dro *sppb.DirectedReadOptions) (bool, error) {
+	if dro != nil {
```

Comment:
Apologies since I am not that familiar with Golang and the correction isn't valid here but isn't it possible to use the function defined in client or to define an independent helper function rather than declaring the exact same function twice?

Reply:
Refactored to use the same function in client for validation DRO.